caddyhttp/httpserver/recorder.go

```diff
@@ -47,6 +47,9 @@ type ResponseRecorder struct {
 // explicitly, this constructor initializes with a status code
 // of 200 to cover the default case.
 func NewResponseRecorder(w http.ResponseWriter) *ResponseRecorder {
+	if rr, ok := w.(*ResponseRecorder); ok {
+		return rr
+	}
```

Comment:
I do wonder what kinds of implications there are of sharing a single ResponseRecorder across all middlewares...

Reply:
Would it be worth providing a test binary with this change in it that people could test out to give us some early indication if it is likely to cause any issues.  I'm not sure there is an answer to this question without extensive testing.